parser/src/lib.rs

```diff
@@ -205,6 +205,35 @@ fn push_literal<'f>(
     Ok(())
 }
 
+/// returns Some(smallest_bit_index, largest_bit_index) contained in `params` if
+///         `params` contains any bitfields.
+///         None otherwise
+pub fn get_max_bitfield_range<'a, I: std::clone::Clone>(params: I) -> Option<(u8, u8)>
```

Comment:
nit: you could move the `: Clone` bound to the where clause below. Also, it shouldn't be necessary to specify the full path of the `Clone` trait; it is always imported by the `core` / `std` [prelude](https://doc.rust-lang.org/core/prelude/v1/index.html).